### ISSUE TITLE
Fine-tuning.ipynb with gpu

### ISSUE THREAD

**Issue by shirishr** [zackchase/mxnet-the-straight-dope#442]
Can some explanation be added to make this notebook run with gpu

Changes like
gpus=1
 and hotdog_w = mx.nd.split(pretrained_weight_param.data().as_in_context(mx.**gpu()**)
        hotdog_b = mx.nd.split(pretrained_bias_param.data().as_in_context(mx.**cpu()**)
and deep_dog_net.load_params('deep-dog-4.params', mx**.gpu()**)
isn't enough

I tried making
new_classifier_w = mx.nd.concat(mx.nd.random_normal(shape=hotdog_w.shape, scale=0.02).**as_in_context(mx.gpu()**, hotdog_w,
                                dim=0)

which also isn't enough.
I wonder is "not_hotdog_validation-c0201740.rec" the culprit?

I get an error at
classify_hotdog(deep_dog_net, '../img/real_hotdog.jpg')
where the second parameter is on cpu. How do I move to gpu?

**Comment by shirishr** [comment]
To run this notebook on gpu...make following changes:
Cell [2]:
change line "gpus = 0" to "gpus = 1"
Cell [7]:
change line "net = models.squeezenet1_1(pretrained=True, prefix='deep_dog_')" to
                    "net = models.squeezenet1_1(pretrained=True, prefix='deep_dog_', ctx = mx.gpu(0))"
Cell [8]
change line "deep_dog_net = models.squeezenet1_1(prefix='deep_dog_', classes=2)" to
                    "deep_dog_net = models.squeezenet1_1(prefix='deep_dog_', classes=2, ctx = mx.gpu(0))"
change line "deep_dog_net.collect_params().initialize()" to
                    "deep_dog_net.collect_params().initialize(ctx = mx.gpu(0))"
Cell[9]:
change line "image = mx.nd.expand_dims(image, axis=0)" to
                    "image = mx.nd.expand_dims(image, axis=0).as_in_context(mx.gpu(0))"

Cell[12]:
change line: "new_classifier_w = mx.nd.concat(mx.nd.random_normal(shape=hotdog_w.shape, scale=0.02),"
to                 "new_classifier_w = mx.nd.concat(mx.nd.random_normal(shape=hotdog_w.shape, scale=0.02).as_in_context(mx.gpu(0)),"

change line "new_classifier_b = mx.nd.concat(mx.nd.random_normal(shape=hotdog_b.shape, scale=0.02)," to
                    "new_classifier_b = mx.nd.concat(mx.nd.random_normal(shape=hotdog_b.shape, scale=0.02).as_in_context(mx.gpu(0)),"

Cell[16]:
change line "deep_dog_net.load_params('deep-dog-4.params', mx.cpu())" to
                   "deep_dog_net.load_params('deep-dog-4.params', mx.gpu())"

change line "deep_dog_net.load_params('deep-dog-5a342a6f.params', mx.cpu())" to
                    "deep_dog_net.load_params('deep-dog-5a342a6f.params', mx.gpu())"
